applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -416,6 +416,134 @@ bool CheckWakeCondition(const Element& rElement, const double& rTolerance, const
     return wake_condition_is_fulfilled;
 }
 
+template <int Dim, int NumNodes>
+double ComputeLocalMachSquaredDerivative(
+    const array_1d<double, Dim>& rVelocity,
+    const double localMachNumber,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
+    //           by Brian Nishida (1996), Section A.2
+
+    // read free stream values
+    const double heat_capacity_ratio = rCurrentProcessInfo[HEAT_CAPACITY_RATIO];
+    const double free_stream_mach = rCurrentProcessInfo[FREE_STREAM_MACH];
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+
+    // make squares of values
+    const double sq_local_mach_number = std::pow(localMachNumber, 2);
+    const double sq_free_stream_mach = std::pow(free_stream_mach, 2);
+    const double sq_free_stream_velocity = inner_prod(free_stream_velocity, free_stream_velocity);
+
+    // computes square of velocity including clamping according to MACH_SQUARED_LIMIT
+    const double sq_local_velocity = ComputeClampedVelocitySquared<Dim, NumNodes>(rVelocity, rCurrentProcessInfo);
+
+    // square bracket term
+    double square_bracket_term = 1.0 + 0.5*(heat_capacity_ratio - 1.0)*
+            sq_free_stream_mach*(1.0 - (sq_local_velocity/sq_free_stream_velocity));
+
+    // calculate mach derivative
+    return sq_local_mach_number * ((1.0/sq_local_velocity) +
+            0.5*(heat_capacity_ratio - 1.0)*(1.0/sq_free_stream_velocity)*sq_free_stream_mach*(1.0/square_bracket_term));
+}
+
+template <int Dim, int NumNodes>
+double ComputeMaximumVelocitySquared(const ProcessInfo& rCurrentProcessInfo)
+{
+    
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
```

Comment:
@inigolcanalejo would one documentation possibility be to make a wiki page on github such as [Constitutive laws in Structural Mechanics Application](https://github.com/KratosMultiphysics/Kratos/wiki/%5BKratosStructuralMechanicsAPI%5D-Constitutive-laws-in-Structural-Mechanics-Application)? Then reference this page in the code?

Reply:
Yes, something like this. But this might bring you a lot of overhead, I would suggest doing this later on and focus now in the implementation to get it done as soon as possible. For now showing the formulas in the PR as images as you did already is more than enough :+1: